dali/operators/reader/loader/filesystem.cc

```diff
@@ -65,15 +67,20 @@ inline void assemble_file_list(std::vector<std::pair<std::string, int>>& file_la
       continue;
     }
 #endif
-    std::string rel_path = curr_entry + dir_sep + std::string{entry->d_name};
-    if (HasKnownExtension(std::string(entry->d_name))) {
-      file_label_pairs.emplace_back(rel_path, label);
+    std::string rel_path = join_path(curr_entry, std::string{entry->d_name});
+    for (auto &filter : filters) {
+      if (fnmatch(filter.c_str(), entry->d_name, case_sensitive_filter ? 0 : FNM_CASEFOLD) == 0) {
```

Comment:
```suggestion
    int flags = case_sensitive_filter ? 0 : FNM_CASEFOLD;
    for (auto &filter : filters) {
      if (fnmatch(filter.c_str(), entry->d_name, flags) == 0) {
```

Reply:
According to the documentation it seems to me that the documentation is not accurate (the part saying that it matches lowercase filters).
I'd double check:
```
filter="*.JPEG"
case_sensitive = false
file="dog.jpeg"
```
^^ this should match. Then the documentation should be updated